sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -498,17 +526,19 @@ impl ObjectRuntimeState {
         let writes: LinkedHashMap<_, _> = transfers
             .into_iter()
             .map(|(id, (owner, type_, value))| {
-                let write_kind =
-                    if input_objects.contains_key(&id) || loaded_child_objects.contains_key(&id) {
-                        debug_assert!(!new_ids.contains_key(&id));
-                        WriteKind::Mutate
-                    } else if id == SUI_SYSTEM_STATE_OBJECT_ID || new_ids.contains_key(&id) {
-                        // SUI_SYSTEM_STATE_OBJECT_ID is only transferred during genesis
-                        // TODO find a way to insert this in the new_ids during genesis transactions
-                        WriteKind::Create
-                    } else {
-                        WriteKind::Unwrap
-                    };
+                let write_kind = if input_objects.contains_key(&id)
```

Comment:
This looks good, but let's just double/triple check the interacts with child objects above. A lot of the states for child objects are inferred based on the transfer states here, and I want to make sure we aren't missing anything.